### PR TITLE
Fixed #425 empty group Ids. Fixed language order in list display

### LINKED ISSUE #425: Empty group identities are considered the same as each other
This causes the templates to be grouped together for list displays, making all but one impossible to see in the list.

### PR DESCRIPTION
**Pull request by seancpeters**

### CODE REVIEW

File: test/Microsoft.TemplateEngine.Core.UnitTests/Microsoft.TemplateEngine.Core.UnitTests.csproj [hunk 1]
```diff
@@ -20,4 +20,7 @@
     <PackageReference Include="xunit" Version="2.2.0" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
```
Comment (prafullbhosale) [inline test/Microsoft.TemplateEngine.Core.UnitTests/Microsoft.TemplateEngine.Core.UnitTests.csproj]: What does this do?

Comment (seancpeters) [inline test/Microsoft.TemplateEngine.Core.UnitTests/Microsoft.TemplateEngine.Core.UnitTests.csproj]: It keeps getting added to test project by VS. We've finally given up on deleting it from each project, each time before check-in.

**Comment by prafullbhosale** [comment]
Do we always merge from default branch to feature branches instead of rebasing in this repo?

**Comment by seancpeters** [comment]
I usually create a branch rebased on rel/vs2017/post-rtw , make my changes, and make a PR. I will also usually rebase against rel/vs2017/post-rtw before making the PR, to avoid merge conflicts. This PR is a little unusual for me, in that I did some merge conflict resolution at the end.

**Pull request merged**